app/features/receive/lightning-address-service.tsx

```diff
@@ -47,6 +50,40 @@ const fakeCryptography: CashuCryptography = {
   },
 };
 
+const sparkMnemonic = process.env.LNURL_SERVER_SPARK_MNEMONIC || '';
+if (!sparkMnemonic) {
+  throw new Error('LNURL_SERVER_SPARK_MNEMONIC is not set');
+}
+
+const queryClient = getQueryClient();
+
+/**
+ * Initializes the singleton Spark wallet for server-side receives.
+ * This should only be called once when the module is first loaded.
+ */
+async function initializeServerSparkWallet() {
```

Comment:
can't we use tanstack query fetchQuery option where queryFn would do SparkWallet.initialize? that way you don't need to manually be setting the cache

Reply:
I am not sure I understand what the bug was